core/src/banking_stage.rs

```diff
@@ -1085,7 +1093,7 @@ impl BankingStage {
                             let poh = poh_recorder.lock().unwrap();
                             poh.bank_start()
                         };
-                        slot_metrics_tracker.update_on_leader_slot_boundary(&current_poh_bank);
+                        slot_metrics_tracker.apply_action(slot_metrics_tracker.check_leader_slot_boundary(&current_poh_bank));
```

Comment:
There's a potential here that we immediately report here, and then don't include the forwarding metrics. 

I think the proper thing here to do is only apply the action if the buffer is nonempty. Otherwise we loop back to the top via a continue

Soemthing like:

```
let metrics_action =  slot_metrics_tracker.check_leader_slot_boundary(&current_poh_bank);
if metrics_action != NoOp {
   if buffered_packet_batches.is_empty() {
      slot_metrics_tracker.apply_action(metrics_action)
   } else {
         // This will cause us to check the forwarding metrics
       continue;
   }
}
```

Reply:
I hesitated to add `continue` as it'd skip `receive_and_buffer_packets()` call once, which changes the behavior a bit. Applied action immediately to keep original behavior (that may miss some fwd counter).

To clarify, here are two cases that adding `continue` would make difference: 
1. Have no active bank and buffer is not empty. `continue` will skip `receiving` once to try to forward those packets in buffer, and count that to prev slot.
2. Have a new bank and buffer is not empty. `continue` skips `receiving` once to process buffered packets, which is counted into new slot.

It is very unlikely, but it is possible (perhaps due to ForwardAndHold) that it could stuck into 1-2-1-2-1 loop therefore not calling `receiving`. wdyt the trade-off between this small risk and accuracy of counter?

Reply:
Hmm how about we simplify this entire loop with this suggested change: https://github.com/solana-labs/solana/commit/df4281bdf77a582f896a06539b0c3955b8ce16f5

Reply:
great idea! apply suggested change